tests/model/test_detector.py

```diff
@@ -305,11 +309,135 @@ def test_project_2d():
     for f1, f2 in zip(f, new_f):
         assert f1 == pytest.approx(f2)
     for s1, s2 in zip(s, new_s):
         assert s1 == pytest.approx(s2)
+
+
+def test_get_panel_projection_2d_from_axes(dials_data):
+
+    from os.path import join
+
+    from dxtbx.model.experiment_list import ExperimentListFactory
+
+    # Get test data
+    pytest.importorskip("h5py")
+    filename = join(
+        dials_data("image_examples"),
+        "SACLA-MPCCD-run266702-0-subset.h5",
+    )
+    experiment = ExperimentListFactory.from_filenames([filename])[0]
+    detector = experiment.detector
+
+    # Get 2d axes
+    origins, fast_axes, slow_axes = get_detector_projection_2d_axes(detector)
+
+    # Get panel 0 data
+    panel = detector[0]
+    image_data = experiment.imageset.get_raw_data(0)[0]
+    fast_axis = matrix.col(fast_axes[0] + (0,))
+    slow_axis = matrix.col(slow_axes[0] + (0,))
+    origin = matrix.col(origins[0] + (0,)) * 1e-3
+
+    # Get 2d projection
+    rotation, translation = get_panel_projection_2d_from_axes(
+        panel, image_data, fast_axis, slow_axis, origin
+    )
+
+    expected_rotation = (
+        -0.0013271608580780825,
+        -0.9999991193216408,
+        0.9999991193216408,
+        -0.0013271608580780825,
+    )
+    expected_translation = (-36.3082159438444, 1033.1228618253663)
+
+    assert len(rotation) == len(expected_rotation)
+    for i, expected_val in enumerate(expected_rotation):
+        assert expected_val == pytest.approx(rotation[i])
+
+    assert len(translation) == len(expected_translation)
+    for i, expected_val in enumerate(expected_translation):
+        assert expected_val == pytest.approx(translation[i])
+
+
+def test_panel_has_projection_2d():
+
+    # Default values given to a panel in panel.h
+    # when a projection_2d is not set
+    empty_rotation = (0, 0, 0, 0)
+    empty_translation = (0, 0)
+
+    detector = create_detector(offset=0)
+    panel = detector[0]
+
+    # Valid projection values
+    valid_rotation = (1, 0, 0, 1)
+    valid_translation = detector[0].get_image_size()
+
+    # Test panel with no projections set explicitly has no 2d projection
+    assert panel.has_projection_2d() is False
+
+    # Test panel with empty 2d projection has no 2d projection
+    panel.set_projection_2d(empty_rotation, empty_translation)
+    assert panel.has_projection_2d() is False
+
+    # Test panel with non-empty 2d projection has 2d projection
+    panel.set_projection_2d(valid_rotation, valid_translation)
+    assert panel.has_projection_2d() is True
+
+
+def test_has_projection_2d():
+
+    # Default values given to a panel when a projection_2d is not set
+    empty_rotation = (0, 0, 0, 0)
+    empty_translation = (0, 0)
+
+    # Valid rotation value
+    rotation = (1, 0, 0, 1)
+
+    ## Test single panel detector
+    detector = create_detector(offset=0)
+
+    # Test detector with no projections set explicity has no 2d projection
+    assert detector.has_projection_2d() is False
```

Comment:
See comments above r.e. explicitly comparing to bool.
```suggestion
    assert not detector.has_projection_2d()
```